ios/RNGeolocation/RNHiveGeolocationManager.swift

```diff
@@ -0,0 +1,323 @@
+//
+//  HiveGeofenceManager.swift
+//  Presence
+//
+//  Created by Andrew on 30/07/2019.
+//  Copyright © 2019 Centrica. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import CoreLocation
+
+public struct RNHiveLocationRequest {
+    let requestId: String
+    let comletion: RNHiveLocationRequestCompletion
+}
+
+enum RNHiveGeofenceCrossingEvent: String {
+    case entry = "ENTER"
+    case exit  = "EXIT"
+}
+
+typealias RNHiveLocationRequestCompletion = (_ locations: [CLLocation]?, _ error: Error?) -> Void
+typealias RNHiveGeofenceRequestCompletion = (_ regions: [CLCircularRegion]?, _ error: Error?) -> Void
+typealias RNHiveGeofenceEventResponder = (_ geofenceEvent: RNHiveGeofenceEvent?, _ error: Error?) -> Void
+
+@objc(RNHiveGeolocationManager)
+class RNHiveGeolocationManager: NSObject {
+    
+    static let shared = RNHiveGeolocationManager()
+    
+    private let locationManager = CLLocationManager()
+    
+    private let savedGeofencesKey = "savedItems"
+    private var geofences: [RNHiveGeofence] = []
+    private var regions: [CLCircularRegion] = []
+    private var pendingLocationRequests: [RNHiveLocationRequest] = []
+    
+    private var geofenceRequestCompletion: RNHiveGeofenceRequestCompletion? = nil
+    private var geofenceEventResponder: RNHiveGeofenceEventResponder? = nil
+    
+    @objc public func allGeofences() -> [RNHiveGeofence] {
+        guard let savedData = UserDefaults.standard.data(forKey: savedGeofencesKey) else { return [] }
+        let decoder = JSONDecoder()
+        if let savedGeotifences = try? decoder.decode(Array.self, from: savedData) as [RNHiveGeofence] {
```

Comment:
presence of `try?` shouldn't be `catch` 
or 
`func allGeofences() throws`?

Reply:
Hmm, I think I would probably agree with you :)